### PR TITLE
[feat] Add ability to create reports

### PR DESCRIPTION
**Pull request by mihran113**

### CODE REVIEW

File: docs/source/ui/pages/reports.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+The following image shows an example of a report created from metrics that were logged to aim over the course of a training. 
+<img alt="View Mode" style="border-radius: 8px; border: 1px solid #E8F1FC" src="https://docs-blobs.s3.us-east-2.amazonaws.com/images/ui/pages/reports/view-mode.png">
+
+
+### Single report page
+In single report page you can find 2 modes: `view` and `edit`. 
+
+
+In `edit` mode you can create the report that you wish for and in view mode see the final result of it, the example shared above is of the `view` mode.
```
Comment (SGevorg) [inline docs/source/ui/pages/reports.md:15]: Create the report in  `edit` mode and view the final result in the `view` mode. The example above is on the `view` mode.

File: docs/source/ui/pages/reports.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+You can use 
+**```aim** sections to retrieve data from BE and visualize them. The example shown below is an overview of the `edit` mode.
```
Comment (SGevorg) [inline docs/source/ui/pages/reports.md:18]: Instead of BE, Aim storage.

File: docs/source/ui/pages/reports.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+<img alt="Edit Mode" style="border-radius: 8px; border: 1px solid #E8F1FC" src="https://docs-blobs.s3.us-east-2.amazonaws.com/images/ui/pages/reports/edit-mode.png">
+
+
+To get acquainted with objects and methods to fetch and visualize data check out the sections below:
```
Comment (SGevorg) [inline docs/source/ui/pages/reports.md:23]: The details on objects and methods for data fetching are in the next section.

File: docs/source/ui/pages/reports.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+### Data fetching methods
+
+To retrieve data from BE the following methods of `repo` object
```
Comment (SGevorg) [inline docs/source/ui/pages/reports.md:27]: Aim Storage

Comment (SGevorg) [reply docs/source/ui/pages/reports.md:27]: Its worth mentioning that `repo` is by default available in the report context.

File: docs/source/ui/pages/reports.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+All the methods accept a query parameter which defaults to `True` (fetches all the available sequences).
+
+The results of this 
+Example:
+
+```python
+```aim
+metrics = repo.fetch_metrics('metric.name == "loss"')
+linechart = LineChart(metrics)
+```\
```
Comment (SGevorg) [inline docs/source/ui/pages/reports.md:46]: Remove the backslash

File: docs/source/ui/pages/reports.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+```
+
+
+### Visualization objects and methods
+
```
Comment (SGevorg) [inline docs/source/ui/pages/reports.md:51]: Its worth adding a few lines here of what these are.

File: docs/source/ui/pages/reports.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+metrics = repo.fetch_metrics()
+linechart = LineChart(metrics)
+linechart.group('color', ['run.hash'])
+linechart.group('row', ['metric_name'])
+```
+
+<img alt="Grouping Example" style="border-radius: 8px; border: 1px solid #E8F1FC" src="https://docs-blobs.s3.us-east-2.amazonaws.com/images/ui/pages/reports/grouping-example.png">
+
+### Search Report 
+You can easily find the report in a big pile with the search ba.
```
Comment (SGevorg) [inline docs/source/ui/pages/reports.md:167]: Typo: bar

Comment (SGevorg) [reply docs/source/ui/pages/reports.md:167]: In general instead I would write:
The search bar is available so you can easily find the report you are looking for.

File: docs/source/ui/pages/reports.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+<img alt="Search Report" style="border-radius: 8px; border: 1px solid #E8F1FC" src="https://docs-blobs.s3.us-east-2.amazonaws.com/images/ui/pages/reports/search-report.png">
+
+### Delete Report
+    
+You can find the `Delete` button under the options button in the upper right corner of the report card.
```
Comment (SGevorg) [inline docs/source/ui/pages/reports.md:172]: The `Delete` button is available with the options...

**Review by SGevorg — CHANGES_REQUESTED** [review]
Left comments for the docs.

**Review by alberttorosyan — APPROVED** [review]
Went over backend changes and docs. BE looks good. Some minor things to be changed in the docs, already covered by Gev's feedback.

**Comment by mihran113** [comment]
> Left comments for the docs.

@SGevorg all the requested changes are addressed. Could you please have a look, and approve?

**Review by SGevorg — APPROVED** [review]
lgtm

**Pull request merged**